### PR TITLE
Update dependencies and switch to Java 11

### PR DESCRIPTION
**Pull request by philipp94831**

### CODE REVIEW

File: large-message-connect/src/test/java/com/bakdata/kafka/AmazonS3IntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/*
```
Comment (torbsto) [inline large-message-connect/src/test/java/com/bakdata/kafka/AmazonS3IntegrationTest.java:1]: We could extract this as a test-fixture into the core module

File: large-message-serde/src/test/java/com/bakdata/kafka/LargeMessageSerializerTest.java [hunk 2]
```diff
@@ -451,9 +362,56 @@ void shouldWriteBackedNullValue() {
...
+        final String deserialized = STRING_DESERIALIZER.deserialize(null, bytes);
+        assertThat(deserialized).isEqualTo(expected);
+    }
+
+    private byte[] readBytes(final BlobStorageURI uri) {
+        try (final InputStream objectContent = this.getS3Client().getObject(GetObjectRequest.builder()
+                .bucket(uri.getBucket())
+                .key(uri.getKey())
+                .build())) {
+            return IoUtils.toByteArray(objectContent);
```
Comment (torbsto) [inline large-message-serde/src/test/java/com/bakdata/kafka/LargeMessageSerializerTest.java:406]:
```suggestion
            return objectContent.readAllBytes();
```

File: large-message-serde/src/test/java/com/bakdata/kafka/LargeMessageSerdeTest.java [hunk 3]
```diff
@@ -114,6 +99,30 @@ void shouldJoin() {
...
+        return properties;
+    }
+
+    private Map<String, Object> getEndpointConfig() {
+        final AwsBasicCredentials credentials = this.getCredentials();
+        final Map<String, Object> largeMessageConfig = new HashMap<>();
+        largeMessageConfig.put(AbstractLargeMessageConfig.S3_ENDPOINT_CONFIG, this.getEndpointOverride().toString());
+        largeMessageConfig.put(AbstractLargeMessageConfig.S3_REGION_CONFIG, this.getRegion().id());
+        largeMessageConfig.put(AbstractLargeMessageConfig.S3_ACCESS_KEY_CONFIG, credentials.accessKeyId());
+        largeMessageConfig.put(AbstractLargeMessageConfig.S3_SECRET_KEY_CONFIG, credentials.secretAccessKey());
```
Comment (torbsto) [inline large-message-serde/src/test/java/com/bakdata/kafka/LargeMessageSerdeTest.java:122]: Maybe move this into the base class?

File: large-message-connect/src/test/java/com/bakdata/kafka/AmazonS3IntegrationTest.java (hunk 1 above)
Comment (philipp94831) [reply large-message-connect/src/test/java/com/bakdata/kafka/AmazonS3IntegrationTest.java:1]: Do you have an example for how to set it up?

Comment (torbsto) [reply large-message-connect/src/test/java/com/bakdata/kafka/AmazonS3IntegrationTest.java:1]: Not really one that is self-containing, but something like this should work:
1. Apply the  `java-test-fixtures` to the core module
2. Move `AmazonS3IntegrationTest` to `large-message-core/src/testFixtures/...`
3. Add `testImplementation(testFixtures(project(":large-message-core")))` to the streams and connect projects

**Review by torbsto — APPROVED** [review]

**Pull request merged**